### PR TITLE
Run all detekt tasks with project dependencies

### PR DESCRIPTION
**Pull request by 3flex**
See https://github.com/arturbosch/detekt/pull/1696#issuecomment-504665839

### CODE REVIEW

File: detekt-sample-extensions/build.gradle.kts [hunk 1]
```diff
@@ -1,13 +1,16 @@
 val assertjVersion: String by project
-val usedDetektVersion: String by project
 val junitPlatformVersion: String by project
 val spekVersion: String by project
 
 dependencies {
-    implementation("io.gitlab.arturbosch.detekt:detekt-api:$usedDetektVersion")
+    // When creating a sample extesion, change this dependency to the detekt-api version you build against, e.g.
```
Comment (schalkms) [inline detekt-sample-extensions/build.gradle.kts]: extension

File: detekt-sample-extensions/build.gradle.kts [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+    // io.gitlab.arturbosch.detekt:detekt-api:1.0.0-RC15
+    implementation(project(":detekt-api"))
     implementation(kotlin("compiler-embeddable"))
 
-    testImplementation("io.gitlab.arturbosch.detekt:detekt-test:$usedDetektVersion")
+    // When creating a sample extesion, change this dependency to the detekt-test version you build against, e.g.
```
Comment (schalkms) [inline detekt-sample-extensions/build.gradle.kts]: extension

**Comment by schalkms** [comment]
Do we need to update the documentation on the website as well?
I mean the section regarding the extension part.

**Comment by 3flex** [comment]
> Why is the test coverage falling down?

Not sure, there's nothing in the PR that should affect coverage at all... possibly Codecov configuration is incorrect or there's something to do with the caching of results that happens between CI runs? For this change only detekt-sample-extensions was rebuilt, the rest of the build would have been cached and reused, so that's probably the reason. Not sure how to fix though.

> Do we need to update the documentation on the website as well?

Good call - I just read through https://arturbosch.github.io/detekt/extensions.html to check. I didn't see anything that would need updating.

**Pull request merged**